pkg/apis/eksctl.io/v1alpha5/validation.go

```diff
@@ -146,6 +146,50 @@ func ValidateNodeGroup(i int, ng *NodeGroup) error {
 		}
 	}
 
+	if err := validateInstancesDistribution(ng); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// TODO What happens with the other instance type. Should we use it as well, ignore it, or fail when both are specified?
+// TODO or should we change it to be an array?
+func validateInstancesDistribution(ng *NodeGroup) error {
+	if ng.InstancesDistribution == nil {
+		return nil
+	}
+
+	if ng.InstanceType != "" && ng.InstanceType != "mixed" {
+		return fmt.Errorf("instanceType should be \"mixed\" or unset when using the mixed instances feature")
+	}
+
+	distribution := ng.InstancesDistribution
+	if distribution.InstanceTypes == nil || len(distribution.InstanceTypes) == 0 {
+		return fmt.Errorf("at least two instance types have to be specified for mixed nodegroups")
+	}
+
+	allInstanceTypes := make(map[string]bool)
+	for _, instanceType := range distribution.InstanceTypes {
+		allInstanceTypes[instanceType] = true
+	}
+
+	if len(allInstanceTypes) < 2 || len(allInstanceTypes) > 20 {
+		return fmt.Errorf("mixed nodegroups should have between 2 and 20 different instance types")
+	}
+
+	if distribution.OnDemandBaseCapacity != nil && *distribution.OnDemandBaseCapacity < 0 {
+		return fmt.Errorf("onDemandBaseCapacity should be 0 or more")
+	}
+
+	if distribution.PercentageAboveBase != nil && (*distribution.PercentageAboveBase < 0 || *distribution.PercentageAboveBase > 100) {
+		return fmt.Errorf("percentageAboveBase should be between 0 and 100")
+	}
+
+	if distribution.SpotInstancePools != nil && (*distribution.SpotInstancePools < 1 || *distribution.SpotInstancePools > 20) {
```

Comment:
API level, SpotInstancePools allows a value greater than `len(allInstanceTypes)`. I am not sure if it makes sense to scope this value here

Reply:
I added this check because I saw that in the documentation it says that the range is 1-20
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-autoscaling-autoscalinggroup-instancesdistribution.html#cfn-autoscaling-autoscalinggroup-instancesdistribution-spotinstancepools
Should it be different?

Reply:
I think this could be an issue on AWS sdk side, if eksctl want to check following case. 
If I use two different instances, I can still have SpotInstancePools > 2 configured here. Do you think this is an issue?

Reply:
I'm not sure I completely understand you, but I think it should be ok because a pool consists of a combination of an instance type and an AZ, so there can be more pools than instance types. 
Here I am only checking that the number is between 1-20 regardless of the number of instance types. Is that wrong?

Reply:
No. It's correct, API doesn't check it neither.